src/ragger/navigator/navigator.py

```diff
@@ -196,26 +203,66 @@ def add_callback(self, ins_id: NavInsID, callback: Callable, override: bool = Tr
                            "registered callbacks")
         self._callbacks[ins_id] = callback
 
-    def navigate(self, instructions: List[Union[NavIns, NavInsID]]):
-        """
-        Navigate on the device according to a set of navigation instructions provided.
+    def _run_instruction(self,
+                         instruction: Union[NavIns, NavInsID],
+                         timeout: float = 10.0,
+                         wait_for_screen_change: bool = True,
+                         path: Optional[Path] = None,
+                         test_case_name: Optional[Path] = None,
+                         snap_idx: int = 0) -> None:
+        if isinstance(instruction, NavInsID):
+            instruction = NavIns(instruction)
+        if instruction.id not in self._callbacks:
+            raise NotImplementedError()
+
+        if instruction.id == NavInsID.USE_CASE_REVIEW_CONFIRM:
+            with NamedTemporaryFile(suffix='.png') as tmp:
+                tmp_file = Path(tmp.name)
+                # Backup screen content before instruction in tmp file
+                self._backend.compare_screen_with_snapshot(tmp_file,
+                                                           tmp_snap_path=tmp_file,
+                                                           golden_run=True)
```

Comment:
Because this is the only way I found so that it don't change too much the Backend interface, and so it stay compatible with ledgercomm and ledgerwallet Backend...

Reply:
Ok why not